### DISCUSSION TITLE
feature: add "dynamic" counterpart to completionCandidates

### DISCUSSION THREAD

**Discussion by alexanderankin** [remkop/picocli#2197] [Ideas]
what if instead of being evaluated at "generate-completions" time, and having the value of the iterable be hardcoded into the bash completion function, but rather the picocli framework would invoke the command and retrieve dynamic values. This is a significant performance tradeoff but may be worth it for some users.

Would like to know if a pr for this would be welcome.

**Comment by remkop** [comment]
Yes, I am open to that. 

Actually I believe there's an older ticket where this idea was explored to some extent. The idea was that a subcommand could generate completions. This was before the GraalVM support was added and at the time it wasn't pursued in depth. I'll try to find that ticket later when I get a chance.

**Reply by alexanderankin** [reply]
Hi Remko, thanks for your reply I will keep researching and prototyping then. Thank you!

**Comment by remkop** [comment]
Please see the discussion in this ticket: https://github.com/remkop/picocli/issues/506
I believe this may be relevant.
